File: Observable/Clima.php
<?php

require_once "./Observable.php";

class Clima extends Observable{
    private $temperatura;

    public function __construct($temperatura)
    {
        $this->temperatura = $temperatura;
    }

    public function atualizarTemp($temperatura){
        $this->temperatura = $temperatura;
        $this->notifyObservers();
    }

    public function getTemperatura(){
        return $this->temperatura;
    }
}

File: Java/observable/Clima.java
package teste;

import java.util.Observable;

public class Clima extends Observable{
	private float temperatura;

	public Clima(float temperatura) {
		super();
		this.temperatura = temperatura;
	}

	public float getTemperatura() {
		return temperatura;
	}

	public void setTemperatura(float temperatura) {
		this.temperatura = temperatura;
		setChanged();
		notifyObservers();
	}
}
